deploy/runtools/firesim_topology_elements.py

```diff
@@ -248,6 +288,12 @@ def run_sim_start_command(self, slotno):
         all_bootbins = [self.get_bootbin_name()]
         all_shmemportnames = [shmemportname]
 
+        assert self.server_hardware_config is not None and isinstance(self.server_hardware_config, RuntimeHWConfig)
```

Comment:
Probably can create a new function called something like `resolved_server_hw_cfg` that has these asserts built-in. Maybe we can have the `server_hw_cfg` be split into `server_hw_cfg` and `server_hw_cfg_name` so that we don't need this?

Reply:
Ok I did this.